### PR TITLE
Fixed sorting for actual price and price columns in subscriptions tab

### PR DESCRIPTION
**Pull request by manan-memon**
### Description:
This pull request addresses an issue with the incorrect sorting of the `Actual Price` and `Price` columns in the Subscriptions tab. The problem arose because DataTables was performing lexicographical sorting instead of numerical sorting, due to an incorrect data type registration for these columns. To resolve this, custom sorting logic has been implemented to ensure accurate numerical sorting.

JIRA: https://edlyio.atlassian.net/browse/EDLY-7141

### Testing Instructions:
1. Sign in to the e-commerce platform.
2. Navigate to the `Subscriptions and Bundles` tab.
3. Test the sorting functionality by clicking the sort buttons on the `Actual Price` and `Price` columns.
4. Verify that the columns now sort correctly in ascending and descending order.

### Screenshot after fix:
image
image

### CODE REVIEW

**Review by Anas-hameed — APPROVED** [review]

File: ecommerce/static/js/views/subscription_list_view.js [hunk 1]
```diff
@@ -19,6 +19,19 @@ define([
...
+         $.extend($.fn.dataTableExt.oSort, {
+            "currency-pre": function(value) {
+                return parseFloat(value.replace(/[^\d.-]/g, '')) || 0;
+            },
+            "currency-asc": function(a, b) {
+                return a - b;
+            },
+            "currency-desc": function(a, b) {
+                return b - a;
+            }
```
Comment (muhammadali286) [inline ecommerce/static/js/views/subscription_list_view.js:32]: nit: Simple arrow function will work
```
"currency-asc": (a, b) => a - b,
"currency-desc": (a, b) => b - a
```

File: ecommerce/static/js/views/subscription_list_view.js [hunk 2]
```diff
@@ -19,6 +19,19 @@ define([
               confirmationDialog) {
         'use strict';
 
+         // Custom sorting plugin for currency values
+         $.extend($.fn.dataTableExt.oSort, {
+            "currency-pre": function(value) {
+                return parseFloat(value.replace(/[^\d.-]/g, '')) || 0;
```
Comment (muhammadali286) [inline ecommerce/static/js/views/subscription_list_view.js:25]: What if extracted value is not a number? 

```
const numericValue = parseFloat(value.replace(/[^\d.-]/g, ''));
return isNaN(numericValue) ? 0 : numericValue;
```

**Review by muhammadali286 — CHANGES_REQUESTED** [review]

File: ecommerce/static/js/views/subscription_list_view.js (hunk 2 above)
Comment (manan-memon) [reply ecommerce/static/js/views/subscription_list_view.js:25]: `parseFloat` will either have a number or an empty string as the regex expression in `value.replace(`) replaces all non-numeric values. Even if it gets a NaN the `|| 0` part will fall back to 0, sorting it accordingly.

File: ecommerce/static/js/views/subscription_list_view.js (hunk 1 above)
Comment (manan-memon) [reply ecommerce/static/js/views/subscription_list_view.js:32]: I just rechecked and found out that we have not used arrow function anywhere in the project. Should I keep it consistent with the repo's standard or replace it with arrow functions?

**Review by muhammadali286 — APPROVED** [review]

**Pull request merged**